packages/dataviews/src/dataform-layouts/normalize-form.ts

```diff
@@ -102,25 +99,44 @@ export function normalizeLayout( layout?: Layout ): NormalizedLayout {
 	return normalizedLayout;
 }
 
-export default function normalizeFormFields(
-	form: Form
-): NormalizedFormField[] {
-	const formLayout = normalizeLayout( form?.layout );
+function normalizeForm( form: Form ): NormalizedForm {
+	const normalizedFormLayout = normalizeLayout( form?.layout );
+
+	const normalizedFields: NormalizedFormField[] = ( form.fields ?? [] ).map(
+		( field ) => {
+			if ( typeof field === 'string' ) {
+				return {
+					id: field,
+					layout: normalizedFormLayout,
+				} satisfies NormalizedFormField;
+			}
+
+			const fieldLayout = field.layout
+				? normalizeLayout( field.layout )
+				: normalizedFormLayout;
 
-	return ( form.fields ?? [] ).map( ( field ) => {
-		if ( typeof field === 'string' ) {
 			return {
-				id: field,
-				layout: formLayout,
-			};
+				id: field.id,
+				layout: fieldLayout,
+				...( !! field.label && { label: field.label } ),
+				...( !! field.description && {
+					description: field.description,
+				} ),
+				...( 'children' in field &&
+					Array.isArray( field.children ) && {
+						children: normalizeForm( {
+							fields: field.children,
+							layout: DEFAULT_LAYOUT,
```

Comment:
Another win: clarifying and add test for how the layout is inherited across nested levels. Only the top-level fields inherit the form layout. Any other layers, are normalized to use the default (regular).